target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-wax620.dts

```diff
@@ -0,0 +1,221 @@
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sck-gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&tlmm 19 GPIO_ACTIVE_HIGH>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <1>;
+			enable-gpios = <&tlmm 20 GPIO_ACTIVE_HIGH>;
+			spi-max-frequency = <1000000>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_system_red: system-red {
+			label = "system:red";
+			gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_system_green: system-green {
+			label = "system:green";
+			gpios = <&tlmm 55 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_system_blue: system-blue {
+			label = "system:blue";
+			gpios = <&tlmm 56 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_lan_g {
+			label = "lan:green";
+			gpios = <&led_gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_lan_o {
+			label = "lan:orange";
+			gpios = <&led_gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_2g_b {
+			label = "wlan2g:blue";
+			gpios = <&led_gpio 2 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		led_2g_g {
+			label = "wlan2g:green";
+			gpios = <&led_gpio 3 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		led_5g_b {
+			label = "wlan5g:blue";
+			gpios = <&led_gpio 4 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		led_5g_g {
+			label = "wlan5g:green";
+			gpios = <&led_gpio 5 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0radio";
+		};
+
+	};
+};
+
+&edma {
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+
+	switch_cpu_bmp = <0x01>;	
+	switch_lan_bmp = <0x3e>;
+	switch_wan_bmp = <0x40>;
+	switch_mac_mode = <0x00>;
+	switch_mac_mode1 = <0xff>;
+	switch_mac_mode2 = <0x0f>;
+	bm_tick_mode = <0x00>;
+	tm_tick_mode = <0x00>;
+
+	qcom,port_phyinfo {
+		port@5 {
+			port_id = <6>;
+			phy_address = <28>;
+			port_mac_sel = "QGMAC_PORT";
+		};
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+};
+
+&mdio {
+	status = "okay";
+
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 44 GPIO_ACTIVE_LOW>;
+
+	qca8081_28: ethernet-phy@28 {
+		reg = <28>;
+	};
+};
+
+&dp6 {
+	status = "okay";
+	phy-handle = <&qca8081_28>;
+	label = "lan";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "okay";
+
+	nand@0 {
+		reg = <0>;
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+		nand-bus-width = <8>;
+
+		partitions {
+			compatible = "qcom,smem-part";
+
+			partition-0-appsblenv {
+				compatible = "fixed-partitions";
+				label = "0:appsblenv";
+				read-only;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partition@0 {
+					compatible = "u-boot,env";
+					label = "env-data";
+					reg = <0x0 0x40000>;
+
+					macaddr_ubootenv_ethaddr: ethaddr {};
```

Comment:
This is unused, is the correct MAC stored there?

If so, you can use it to set the ethernet MAC instead of relying on U-Boot.

Reply:
From printenv I can see the following:
`ethaddr=80:cc:9c:57:e0:5f`

That means it's loading from the uboot right? Do I have to delete a line, or add something?

Reply:
If u-boot can't patch the MAC-address during boot, ssdk should print a message during init regarding wrong mac-address. (I don't see such an error in your provided bootlog https://gist.github.com/skramstad/50ab66f5aba1fc31e5a3f3f9127c3c5a)

You can double check that by "ip a" and look if the lan interface does have the same ethaddr from printenv.

If so, you can remove the entire partition-0-appsblenv definition from the dts.

Reply:
Yes, you are correct @kirdesde. The lan interface has the same ethaddr from printenv.
So I just remove all of this?
```
partition@0 {
	compatible = "u-boot,env";
	label = "env-data";
	reg = <0x0 0x40000>;

	macaddr_ubootenv_ethaddr: ethaddr {};
};
```

Reply:
You can remove this block:

```
partition-0-appsblenv {
				compatible = "fixed-partitions";
				label = "0:appsblenv";
				read-only;
				#address-cells = <1>;
				#size-cells = <1>;

				partition@0 {
					compatible = "u-boot,env";
					label = "env-data";
					reg = <0x0 0x40000>;

					macaddr_ubootenv_ethaddr: ethaddr {};
				};
			};
```